From 79c8c73949f51359dca40ba1b9286c866d9844cf Mon Sep 17 00:00:00 2001
From: Behnam Heydarshahi <bhnm@google.com>
Date: Tue, 8 Apr 2025 13:59:15 -0700
Subject: [PATCH] Unexport ControlsActivity

Also do not draw non-system windows on top of ControlsSettingsDialog

Fix: 404256832
Flag: EXEMPT bugfix
Test: atest ControlsSettingsDialogManagerImplTest

Change-Id: I6280b501b799596099935dc61ecef0b431dd8333
Merged-In: I63e98794e1a93e3c208ee421654d73542d2bd34a
---
 packages/SystemUI/AndroidManifest.xml             |  1 -
 .../settings/ControlsSettingsDialogManager.kt     |  3 +++
 .../ControlsSettingsDialogManagerImplTest.kt      | 15 +++++++++++++++
 3 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/AndroidManifest.xml b/packages/SystemUI/AndroidManifest.xml
index 5c036988eb30..2036cc415162 100644
--- a/packages/SystemUI/AndroidManifest.xml
+++ b/packages/SystemUI/AndroidManifest.xml
@@ -949,7 +949,6 @@
                   android:launchMode="singleInstance"
                   android:configChanges="screenSize|smallestScreenSize|screenLayout|keyboard|keyboardHidden|orientation"
                   android:visibleToInstantApps="true"
-                  android:exported="true"
                   />
 
         <activity android:name=".wallet.ui.WalletActivity"
diff --git a/packages/SystemUI/src/com/android/systemui/controls/settings/ControlsSettingsDialogManager.kt b/packages/SystemUI/src/com/android/systemui/controls/settings/ControlsSettingsDialogManager.kt
index ce0f2e93f26d..7e114557e896 100644
--- a/packages/SystemUI/src/com/android/systemui/controls/settings/ControlsSettingsDialogManager.kt
+++ b/packages/SystemUI/src/com/android/systemui/controls/settings/ControlsSettingsDialogManager.kt
@@ -23,6 +23,7 @@ import android.content.Context.MODE_PRIVATE
 import android.content.DialogInterface
 import android.content.SharedPreferences
 import android.provider.Settings
+import android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS
 import androidx.annotation.VisibleForTesting
 import com.android.systemui.R
 import com.android.systemui.controls.settings.ControlsSettingsDialogManager.Companion.MAX_NUMBER_ATTEMPTS_CONTROLS_DIALOG
@@ -148,6 +149,8 @@ internal constructor(
                 }
             }
 
+        d.window?.addSystemFlags(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS)
+
         SystemUIDialog.registerDismissListener(d) { dialog = null }
         SystemUIDialog.setDialogSize(d)
         SystemUIDialog.setShowForAllUsers(d, true)
diff --git a/packages/SystemUI/tests/src/com/android/systemui/controls/settings/ControlsSettingsDialogManagerImplTest.kt b/packages/SystemUI/tests/src/com/android/systemui/controls/settings/ControlsSettingsDialogManagerImplTest.kt
index 590989d3a987..d59a9d999a24 100644
--- a/packages/SystemUI/tests/src/com/android/systemui/controls/settings/ControlsSettingsDialogManagerImplTest.kt
+++ b/packages/SystemUI/tests/src/com/android/systemui/controls/settings/ControlsSettingsDialogManagerImplTest.kt
@@ -24,6 +24,8 @@ import android.provider.Settings.Secure.LOCKSCREEN_ALLOW_TRIVIAL_CONTROLS
 import android.provider.Settings.Secure.LOCKSCREEN_SHOW_CONTROLS
 import android.testing.AndroidTestingRunner
 import android.testing.TestableLooper
+import android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS
+import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.filters.SmallTest
 import com.android.systemui.SysuiTestCase
 import com.android.systemui.controls.settings.ControlsSettingsDialogManager.Companion.PREFS_SETTINGS_DIALOG_ATTEMPTS
@@ -327,6 +329,19 @@ class ControlsSettingsDialogManagerImplTest : SysuiTestCase() {
         assertThat(secureSettings.getBool(SETTING_ACTION)).isTrue()
     }
 
+    @Test
+    fun dialogHidesNonSystemOverlayWindows() {
+        sharedPreferences.putAttempts(0)
+        secureSettings.putBool(SETTING_SHOW, false)
+        secureSettings.putBool(SETTING_ACTION, false)
+
+        underTest.maybeShowDialog(context, completedRunnable)
+
+        val flags = dialog!!.window!!.attributes.privateFlags
+        assertThat(flags and SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS)
+            .isEqualTo(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS)
+    }
+
     private fun clickButton(which: Int) {
         dialog?.clickButton(which)
     }
-- 
2.50.1.351.gcf940e82a1-goog

